### PR TITLE
ENH: Simplify filter_type in *ord() functions

### PR DESCRIPTION
**Pull request by endolith**
#### What does this implement/fix?
Calculation of filter_type can be simplified, using the same calculation in each function, and catch invalid filter types early

### CODE REVIEW

File: scipy/signal/filter_design.py [hunk 1]
```diff
@@ -3524,8 +3524,8 @@ def buttord(wp, ws, gpass, gstop, analog=False, fs=None):
         wp = 2*wp/fs
         ws = 2*ws/fs
 
-    filter_type = 2 * (len(wp) - 1)
-    filter_type += 1
+    # 1 = lp, 2 = hp, 3 = bs, 4 = bp
+    filter_type = 2 * len(wp) - 1
```
Comment (tylerjereddy) [inline scipy/signal/filter_design.py]: This case seems straightforward to understand the simplification for

File: scipy/signal/filter_design.py [hunk 2]
```diff
@@ -3692,11 +3692,10 @@ def cheb1ord(wp, ws, gpass, gstop, analog=False, fs=None):
         wp = 2*wp/fs
         ws = 2*ws/fs
 
-    filter_type = 2 * (len(wp) - 1)
-    if wp[0] < ws[0]:
+    # 1 = lp, 2 = hp, 3 = bs, 4 = bp
+    filter_type = 2 * len(wp) - 1
```
Comment (tylerjereddy) [inline scipy/signal/filter_design.py]: This "simplification" looks a bit less obvious to me?

File: scipy/signal/filter_design.py [hunk 3]
```diff
@@ -3557,6 +3557,8 @@ def buttord(wp, ws, gpass, gstop, analog=False, fs=None):
     elif filter_type == 4:          # pass
         nat = ((stopb ** 2 - passb[0] * passb[1]) /
                (stopb * (passb[0] - passb[1])))
+    else:
+        raise ValueError("Bad filter type: %s" % filter_type)
```
Comment (tylerjereddy) [inline scipy/signal/filter_design.py]: Perhaps worth noting that this code path is never hit in the test suite, for this or any of the other similar `ValueError` entries adjusted in this PR.

**Review by tylerjereddy — CHANGES_REQUESTED** [review]
We probably don't *have* to add more tests here, but using "request changes" so the PR gets properly tracked by Pauli's system.

Maybe good just to clarify the slightly-less-obvious streamlining case.

File: scipy/signal/filter_design.py (hunk 3 above)
Comment (endolith) [reply scipy/signal/filter_design.py]: That's a good point

File: scipy/signal/filter_design.py (hunk 2 above)
Comment (endolith) [reply scipy/signal/filter_design.py]: I made it consistent with the other one, which is also simpler.  

len(wp) is either 1 or 2 (or invalid)

filter_type = 2 * (len(wp) - 1)
is basically {1: 0, 2: 2}
then
if wp<ws it becomes {1: 1, 2: 3}
if wp>ws it becomes {1: 2, 2: 4}

filter_type = 2 * len(wp) - 1
is basically {1: 1, 2: 3}
which is already correct for wp\<ws 
then
if wp>ws it becomes {1: 2, 2: 4}

File: scipy/signal/filter_design.py (hunk 3 above)
Comment (larsoner) [reply scipy/signal/filter_design.py]: Probably worth adding some `with pytest.raises(ValueError, match='Bad filter type'):` to the tests

**Comment by ev-br** [comment]
Apparently almost this was done in https://github.com/scipy/scipy/pull/18670 : cf 

https://github.com/scipy/scipy/blob/main/scipy/signal/_filter_design.py#L3701 and https://github.com/scipy/scipy/blob/main/scipy/signal/_filter_design.py#L3836

I'm going to close this PR then, thank you regardless @endolith !  --- if you see an issue with gh-18760, please comment (or fix).

**Pull request closed without merging**